core/src/main/scala/com/criteo/cuttle/platforms/WaitingExecutionQueue.scala

```diff
@@ -53,10 +53,13 @@ trait WaitingExecutionQueue {
     }
     execution
       .onCancel(() => {
-        atomic { implicit txn =>
+        // we cancel only waiting executions because the running ones are not in responsibility of the queue.
```

Comment:
I am not sure this is related to this PR. I'll leave it up to you if you really want to merge this here.